### ISSUE TITLE
BCH Decode of BCH(255,207, t=6) failing

### ISSUE THREAD

**Issue by caronoff** [jkent/python-bchlib#24]
Our organization uses a shortened form of a (255,207, t=6) Bose-Chaudhuri-Hocquenghem (BCH) code. The shortened form (250,202) consists of 202 bits of data followed by a 48-bit sextuple error-correcting code.   The Python library wrapper does not result in any ECC less than 56 bits.  

We need a wrapper in Python that can be used to detect and correct up to 6 errors in the entire 250-bit pattern (bits 1 through 250 of the message).
Note: For the purpose of error correction, all calculations shall be performed with the full 255 length code. Therefore, 5 zeros are placed before the 202 data bits to form the 207-bit pattern of the (255,207) BCH code. These padding zeros do not affect the generation of the BCH code as described below.
For the (250,202) BCH code, a generator polynomial g(X) (the same as for (255,207) BCH code) is defined as follows:
𝑔(𝑋)=𝐿𝐶𝑀(𝑚1(𝑋),𝑚3(𝑋),𝑚5(𝑋),𝑚7(𝑋),𝑚9(𝑋),𝑚11(𝑋))

shortened form of a (255,207) Bose-Chaudhuri-Hocquenghem (BCH) code. The shortened form (250,202) consists of 202 bits of data followed by a 48-bit sextuple error-correcting code. The code is used to detect and correct up to six errors in the entire 250-bit pattern (bits 1 through 250 of the 406 MHz message).
Note: For the purpose of error correction, all calculations shall be performed with the full 255 length code. Therefore, 5 zeros are placed before the 202 data bits to form the 207-bit pattern of the (255,207) BCH code. These padding zeros do not affect the generation of the BCH code as described below.
For the (250,202) BCH code, a generator polynomial g(X) (the same as for (255,207) BCH code) is defined as follows:
𝑔(𝑋)=𝐿𝐶𝑀(𝑚1(𝑋),𝑚3(𝑋),𝑚5(𝑋),𝑚7(𝑋),𝑚9(𝑋),𝑚11(𝑋))

**Comment by caronoff** [comment]
Is anyone maintaining this still?

It should be able to correct 6 errors in a BCH(255,207,m=8, t=6).  The g(x) should  be determined by:

𝑔(𝑋)=𝐿𝐶𝑀(𝑚1(𝑋),𝑚3(𝑋),𝑚5(𝑋),𝑚7(𝑋),𝑚9(𝑋),𝑚11(𝑋))
where LCM = Least Common Multiple.
In the above case:

𝑚1(𝑋)= 𝑋^8+𝑋^4+𝑋^3+𝑋^2+1
𝑚3(𝑋)= 𝑋^8+𝑋^6+𝑋^5+𝑋^4+𝑋^2+𝑋+1
𝑚5(𝑋)= 𝑋^8+𝑋^7+𝑋^6+𝑋^5+𝑋^4+𝑋+1
𝑚7(𝑋)= 𝑋^8+𝑋^6+𝑋^5+𝑋^3+1
𝑚9(𝑋)= 𝑋^8+𝑋^7+𝑋^5+𝑋^4+𝑋^3+𝑋^2+1
𝑚11(𝑋)= 𝑋^8+𝑋^7+𝑋^6+𝑋^5+𝑋^2+𝑋+1

This is in binary notation:   111100111
The primative polynolial value is :  487

From which we should get G(x) of 
𝑔(𝑋)
=𝑚1(𝑋),𝑚3(𝑋),𝑚5(𝑋),𝑚7(𝑋),𝑚9(𝑋),𝑚11(𝑋)=𝑋48+𝑋47+𝑋46+𝑋42+𝑋41+𝑋40+𝑋39+𝑋38+𝑋37+𝑋35+𝑋33+𝑋32+𝑋31+𝑋26+𝑋24+𝑋23+𝑋22+𝑋20+𝑋19+𝑋18+𝑋17+𝑋16+𝑋13+𝑋12+𝑋11+𝑋10+𝑋7+𝑋4+𝑋2+𝑋+1

𝑔(𝑋)=1110001111110101110000101110111110011110010010111

https://web.ntpu.edu.tw/~yshan/BCH_code.pdf (see page 8)

**Comment by caronoff** [comment]
Hi Jeff,
Not sure if you're following my notes but I will keep updating you as I can...

On the current, non testing build, I am able to use the BCH decoder with the bch.decode_inplace(data,ecc).  
My constructor is:
    BCH_POLYNOMIAL = 285
    BCH_BITS = 6    
    bch = bchlib.BCH(BCH_POLYNOMIAL, BCH_BITS)

The maximum size of the data that can be decoded_in place is still only 25 bytes (200 bits).  However, for the encode, the ECC can be computed on > 25 bytes, up to 207 bits of 'data'  produces  the ecc.
  ecc = bch.encode(data)

Once you get down to the correction, I need to start at the SECOND byte to form the data.

data, ecc = packet[1:-bch.ecc_bytes], packet[-bch.ecc_bytes:]

this is the only way the command decode_inplace will work!  Data bytearray must be 25 bytes maximum.
bch.decode_inplace(data,ecc).

In summary, I can correct up to 6 bits of errors only within the 200 bits of data + 48 bits.   7 bits at the beginning are lost from error correction.   This is the same as with the new testing build.

Therefore, there we need to find a way for data size to be able to take 26 bytes of data for correction step.
 ( or bch.correct(data,ecc) in the new build)
